packages/commerce-sdk-react/src/hooks/useQuery.ts

```diff
@@ -66,3 +70,64 @@ export const useQuery = <Client extends ApiClient, Options extends ApiOptions, D
             : {})
     })
 }
+
+/**
+ * A hook for SCAPI custom endpoint queries.
+ *
+ * Besides calling custom endpoint, this hook does a few things for better DX.
+ * 1. inject access token
+ * 2. merge SCAPI client configurations from the CommerceApiProvider
+ * @param apiOptions - Options passed through to commerce-sdk-isomorphic
+ * @param queryOptions - Options passed through to @tanstack/react-query
+ * @returns A TanStack Query query hook with data from the custom API endpoint.
+ */
+export const useCustomQuery = (
+    apiOptions: CustomEndpointArgClientConfigOptional,
+    queryOptions?: UseQueryOptions<unknown, unknown, unknown, any>
+) => {
+    const config = useConfig()
+    const auth = useAuthContext()
+    const callCustomEndpointWithAuth = (options: CustomEndpointArgClientConfigOptional) => {
+        const clientConfig = options.clientConfig || {}
+        const clientHeaders = config.headers || {}
+        return async () => {
+            const {access_token} = await auth.ready()
+            return await helpers.callCustomEndpoint({
+                ...options,
```

Comment:
same as below comment

Reply:
it's intentional since we further deconstruction `...options.options` and `...options.clientConfig`, so it's all overridable but just not the clientConfig parameters and proxy